flag-app/flag-app-project/src/components/Search.tsx

```diff
@@ -0,0 +1,31 @@
+import { useRef } from "react";
+import { IoMdSearch } from "react-icons/io";
+
+interface Props {
+  onSearch: (searchText: string) => void;
+}
+
+const Search = ({ onSearch }: Props) => {
+  const ref = useRef<HTMLInputElement>(null);
+
+  return (
+    <form
+      onSubmit={(event) => {
+        event.preventDefault();
+        if (ref.current) onSearch(ref.current.value);
+      }}
```

Comment:
This onSubmit handler should be separated into a function. If there were 20 more lines, this would be harder to read, so it's always better to make this a named function and call it here.